### PR TITLE
Span.LineBreak

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

```xaml
<LineBreak/>
```

### API Changes ###

 
 None

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Core/Span.cs [hunk 1]
```diff
@@ -55,7 +55,8 @@ public Color ForegroundColor
 #pragma warning restore 618
 
 		public static readonly BindableProperty TextProperty
-			= BindableProperty.Create(nameof(Text), typeof(string), typeof(Span), "", defaultBindingMode: BindingMode.OneWay);
+			= BindableProperty.Create(nameof(Text), typeof(string), typeof(Span), "", defaultBindingMode: BindingMode.OneWay,
+				validateValue: (b, v) => !(b is NewLine));
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Span.cs]:
```suggestion
				validateValue: (b, v) => !(b is LineBreak));
```

File: Xamarin.Forms.Core/Span.cs [hunk 2]
```diff
@@ -159,4 +160,9 @@ void ILineHeightElement.OnLineHeightChanged(double oldValue, double newValue)
 		{
 		}
 	}
+
+	public class NewLine : Span
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Span.cs]:
```suggestion
	public class LineBreak : Span
```

File: Xamarin.Forms.Core/Span.cs [hunk 3]
```diff
@@ -159,4 +160,9 @@ void ILineHeightElement.OnLineHeightChanged(double oldValue, double newValue)
...
+	{
+		public NewLine() => Text = Environment.NewLine;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Span.cs]:
```suggestion
		public LineBreak() => Text = Environment.NewLine;
```

**Comment by dotMorten** [comment]
> This was a quick and dirty implementation for david, and it ended up being mostly dirty...

At least you're honest with yourself :-) Should Xamarin.Forms be adding dirty stuff though? I mean the above implementation anyone could do in their own library. IMHO the Xamarin.Forms implementation should be solid for the long run.

Btw I missed you making it not modifiable with that validate value callback. Very clever trick/hack! :-)

**Comment by StephaneDelcroix** [comment]
@dotMorten I'm not going to merge this as is

**Comment by KSemenenko** [comment]
Maybe add something like <Space /> which will insert " " character?

**Comment by adrianknight89** [comment]
@StephaneDelcroix I don't see a problem with making this a special `Span` as long as the validation is in place. Creating two different types could be a breaking change. For example, `FormattedText.Spans` collection will need to be renamed and we can't add a `LineBreak` to it unless the underlying data type for the collection is changed. This is too much work for this tiny element.

My only issue is that line breaks are often not semantic. See [here](https://stackoverflow.com/questions/1726073/is-it-sometimes-bad-to-use-br).

**Pull request closed without merging**